### PR TITLE
Clean up spline2 folder

### PR DESCRIPTION
**Pull request by ye-luo**
There were a few experimental version of MultiSpline evaluation routines.
It has been reduced to keep only the one in use.
Add a bit description and safeguard in MultiBspline.hpp

### CODE REVIEW

File: src/spline2/MultiBsplineStd.hpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace qmcplusplus
   template<typename T>
     inline void 
     MultiBspline<T>::evaluate_vgl_impl(T x, T y, T z, 
-        T* restrict vals, T* restrict grads, T* restrict lapl, int first, int last, size_t out_offset) const
+        T* restrict vals, T* restrict grads, T* restrict lapl, size_t out_offset) const
```
Comment (markdewing) [inline src/spline2/MultiBsplineStd.hpp]: out_offset was changed to out_stride elsewhere

Comment (markdewing) [reply src/spline2/MultiBsplineStd.hpp]: Nevermind.  I see it really is an offset here.

Comment (ye-luo) [reply src/spline2/MultiBsplineStd.hpp]: Let me check. I probably changed one place but forgot another. What is a better out_offset or out_stride?

Comment (ye-luo) [reply src/spline2/MultiBsplineStd.hpp]: Changed back to out_offset.

Comment (markdewing) [reply src/spline2/MultiBsplineStd.hpp]: out_stride implies the fastest dimension, at least to me.  out_offset seems better, as it is the distance between the SoA elements.

**Pull request merged**